### PR TITLE
Add support for building contexts objects

### PR DESCRIPTION
**Pull request by moysesb**
This PR implements #538 and is necessary for docker-compose PR https://github.com/docker/compose/pull/1244

### CODE REVIEW

File: docker/utils/context.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        pass
+
+
+class ContextError(Exception):
+    def __init__(self, msg):
+        self.message = msg
+
+
+INVALID_CONTEXT_FORMAT_LONG_FMT = """
+Build context at %s is not supported by docker-compose\\n
```
Comment (aanand) [inline docker/utils/context.py]: This should probably just say "Docker" :smirk:

File: docker/utils/context.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+REMOTE_CONTEXT_PREFIXES = ["http://",
+                           "https://",
+                           "git://",
+                           "git@",
+                           "github.com/"]
+
+DOCKERFILE_CMD_RG = re.compile(r"[\t\s]*\\n$|([\s\t]*"
+                               r"FROM|MAINTAINER|RUN|CMD|EXPOSE|ENV|ADD|COPY|"
+                               r"ENTRYPOINT|VOLUME|USER|WORKDIR|ONBUILD"
+                               r"[\s\t]+.+)")
```
Comment (aanand) [inline docker/utils/context.py]: What's this constant used for?

Comment (moysesb) [inline docker/utils/context.py]: Removed now. It would eventually be used for validating a Dockerfile thoroughly, but I discarded the idea.

**Comment by aanand** [comment]
1. How much of this logic is actually implemented in the Docker client? For example, validation of the Dockerfile seems to happen on the server side. We don't need to go beyond what the Docker client implements.

2. Documentation would be great.

**Comment by moysesb** [comment]
@aanand 
1.  Scaled back the validation to be just on access rights and Dockerfile name check. The docker go client does roughly that level of validation, but since it supports reading a compressed context from stdin it implicitly rejects a malformed tarball.
2.  Added documentation at [docs/context.md] (https://github.com/moysesb/docker-py/blob/distinct_build_calls/docs/context.md)

**Comment by aanand** [comment]
Thanks. Looking good.

**Pull request closed without merging**